File: __tests__/data_structures/linked_list_test.py
from logging import setLoggerClass
import sys
sys.path.insert(0, '/home/kayode/Documents/Learning/interview_questions/data_structures')
import unittest
from linked_list import SinglyLinkedList

class LinkedListTestCases(unittest.TestCase):
    def setUp(self):
        self.linkedlist = SinglyLinkedList("first")
        self.linkedlist.add("second")

    def test_second_item_present(self):
        self.assertEqual(self.linkedlist.search("second").data, "second")

    def test_size_of_linkedlist(self):
        self.linkedlist.add("third")
        self.linkedlist.add("fourth")
        self.assertEqual(self.linkedlist.size(), 4)
        self.linkedlist.add("fifth")
        self.assertEqual(self.linkedlist.size(), 5)

    def test_remove_item(self):
        self.linkedlist.remove("second")
        self.assertEqual(self.linkedlist.size(), 1)


if __name__ == '__main__':    
    unittest.main()

File: __tests__/data_structures/linkedlist.test.js
const { SinglyLinkedList } = require("../../data_structures/linked_list");

describe("test the implemetation of linkedlist in JS", () => {
  const ll = new SinglyLinkedList();
  ll.add("1");
  ll.add("2");
  ll.add("3");

  test("list contains items", () => {
    expect(ll.print()).toEqual("3 2 1");
  });

  test("remove items", () => {
    ll.remove("2");
    expect(ll.print()).toEqual("3 1");
  });
});
